src/spikeinterface/core/recording_tools.py

```diff
@@ -2,9 +2,486 @@
+    )
+    traces = traces.astype(dtype, copy=False)
+    arr[start_frame:end_frame, :] = traces
+
+
+def write_memory_recording(recording, dtype=None, verbose=False, auto_cast_uint=True, **job_kwargs):
+    """
+    Save the traces into numpy arrays (memory).
+    try to use the SharedMemory introduce in py3.8 if n_jobs > 1
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor object to be saved in .dat format
+    dtype: dtype, default: None
+        Type of the saved data
+    verbose: bool, default: False
+        If True, output is verbose (when chunks are used)
+    auto_cast_uint: bool, default: True
+        If True, unsigned integers are automatically cast to int if the specified dtype is signed
+    {}
+
+    Returns
+    ---------
+    arrays: one arrays per segment
+    """
+    job_kwargs = fix_job_kwargs(job_kwargs)
+
+    if dtype is None:
+        dtype = recording.get_dtype()
+    if auto_cast_uint:
+        cast_unsigned = determine_cast_unsigned(recording, dtype)
+    else:
+        cast_unsigned = False
+
+    # create sharedmmep
+    arrays = []
+    shm_names = []
+    shapes = []
+
+    n_jobs = ensure_n_jobs(recording, n_jobs=job_kwargs.get("n_jobs", 1))
+    for segment_index in range(recording.get_num_segments()):
+        num_frames = recording.get_num_samples(segment_index)
+        num_channels = recording.get_num_channels()
+        shape = (num_frames, num_channels)
+        shapes.append(shape)
+        if n_jobs > 1:
+            arr, shm = make_shared_array(shape, dtype)
+            shm_names.append(shm.name)
+        else:
+            arr = np.zeros(shape, dtype=dtype)
+        arrays.append(arr)
+
+    # use executor (loop or workers)
+    func = _write_memory_chunk
+    init_func = _init_memory_worker
+    if n_jobs > 1:
+        init_args = (recording, None, shm_names, shapes, dtype, cast_unsigned)
+    else:
+        init_args = (recording, arrays, None, None, dtype, cast_unsigned)
+
+    executor = ChunkRecordingExecutor(
+        recording, func, init_func, init_args, verbose=verbose, job_name="write_memory_recording", **job_kwargs
+    )
+    executor.run()
+
+    return arrays
+
+
+write_memory_recording.__doc__ = write_memory_recording.__doc__.format(_shared_job_kwargs_doc)
+
+
+def write_to_h5_dataset_format(
+    recording,
+    dataset_path,
+    segment_index,
+    save_path=None,
+    file_handle=None,
+    time_axis=0,
+    single_axis=False,
+    dtype=None,
+    chunk_size=None,
+    chunk_memory="500M",
+    verbose=False,
+    auto_cast_uint=True,
+    return_scaled=False,
+):
+    """
+    Save the traces of a recording extractor in an h5 dataset.
+
+    Parameters
+    ----------
+    recording: RecordingExtractor
+        The recording extractor object to be saved in .dat format
+    dataset_path: str
+        Path to dataset in h5 file (e.g. "/dataset")
```

Comment:
```suggestion
        Path to dataset in h5 file (e.g. "/dataset")
```

I don't know this format as well. This should be `for` unless you are trying to say that the path exists inside the name?

Reply:
Hi Zach, I don't understand the comment! Ot seems you are not changing anything in the suggesiton

Reply:
Hey Alessio (and Happy New Year by the way!) I was saying that the standard way to say this is:

Path to dataset __for__ the h5 file (e.g. "/dataset"), 

but that since I don't know this format maybe you meant that the name was inside of the file path like:

(e.g."nameofdata/dataset") in which case the using the word `in` is okay. Since I never use this format I didn't want to correct the preposition if that preposition was actually import for the use of the parameter. 

Does that make sense? (ie typically the preposition used is wrong unless specifically meaning that the Path is inside of the h5 file format somewhere.

Reply:
Happy new year to you too! 

Got it! I think that in this case "Path to dataset **in** the h5 file (e.g. "/dataset")" would be the best :)

Reply:
done :)